src/acceptance/java/io/codurance/pouch/FavouriteStatusFeature.java

```diff
@@ -0,0 +1,93 @@
+package io.codurance.pouch;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static io.restassured.RestAssured.when;
+import static java.time.Instant.now;
+import static java.util.UUID.randomUUID;
+import static org.apache.http.HttpStatus.SC_NOT_FOUND;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.hamcrest.Matchers.isEmptyString;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = PouchApiApplication.class, webEnvironment = WebEnvironment.RANDOM_PORT)
+public class FavouriteStatusFeature {
+
+    @LocalServerPort
+    private int serverPort;
+
+    @Value("${spring.datasource.jdbc-url}")
+    private String jdbcUrl;
+
+    private DatabaseHelper databaseHelper;
+
+    @Before
+    public void setUp() {
+        databaseHelper = new DatabaseHelper(jdbcUrl);
+        databaseHelper.clearResources();
+
+        RestAssuredConfiguration.configure(serverPort);
+    }
+
+    @Test
+    public void shouldAddFavouriteStatusToResource() {
+        var randomUUID = randomUUID();
+        var currentTimestamp = now();
+        var resourceToFavourite = new DatabaseHelper.Resource(
+                randomUUID,
+                currentTimestamp,
+                "Hacker Noon",
+                "https://hackernoon.com/");
+
+        databaseHelper.insertResource(resourceToFavourite);
+
+        when().put("/resources/" + randomUUID.toString() + "/favourite")
+                .then()
+                .statusCode(SC_OK);
+    }
+
+    @Test
+    public void shouldRemoveFavouriteStatusOnResource() {
+        var randomUUID = randomUUID();
+        var currentTimestamp = now();
+        var resourceToFavourite = new DatabaseHelper.Resource(
+                randomUUID,
+                currentTimestamp,
+                "Hacker Noon",
+                "https://hackernoon.com/");
+
+        resourceToFavourite.setFavourite(true);
+        databaseHelper.insertResource(resourceToFavourite);
+
+        when().delete("/resources/" + randomUUID.toString() + "/favourite")
+                .then()
+                .statusCode(SC_OK);
```

Comment:
I think the correct status code if no data is returned should be no content
9.7 DELETE

A successful response SHOULD be 200 (OK) if the response includes an entity describing the status, 202 (Accepted) if the action has not yet been enacted, or 204 (No Content) if the action has been enacted but the response does not include an entity.

Reply:
What status code to return in any given situation (or rather...at least in the most common situations) is still not obvious to me. So far, @alfredodev pointed me towards the [REST API v3](https://developer.github.com/v3/activity/starring/) and @andretorres-codurance recommended to use [Rails Routing from the Outside In](https://guides.rubyonrails.org/routing.html) from the (Ruby on) RailsGuides.

Reply:
I changed the status code being returned to `204 NO CONTENT`.